### PR TITLE
Add availability related API

### PR DESCRIPTION
**Pull request by G-Salviati**
This PR adds three API defined in this [task](https://polito-se2-21-09.myjetbrains.com/youtrack/issue/S202109SPG-78) and update the findProducts API in order to show only available products

### CODE REVIEW

File: server/dao/product.js [hunk 1]
```diff
@@ -37,6 +38,47 @@ exports.findProducts = (db, searchString, category) => {
...
+    query = {
+      $and: [
+        { farmerID: ObjectID(farmerID) },
+        { $text: { $search: searchString.toString() } },
+      ],
+    };
+  } else {
+    query = { farmerID: ObjectID(farmerID) };
+  }
+
```
Comment (MrGreensushi) [inline server/dao/product.js:71]: to create the query this way is a little bit confusing.
```
query="{$and:[ {farmerID:ObjectID(famrerID)},";
if(category)
query+="{ category: category.toString() },";
```
etc

File: server/dao/product.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 "use strict";
 
 const { ObjectID } = require("bson");
+const { productsCollection } = require("../test/test-data");
```
Comment (MrGreensushi) [inline server/dao/product.js]: What's the use of this here?

**Review by MrGreensushi — CHANGES_REQUESTED** [review]

File: server/dao/product.js (hunk 1 above)
Comment (G-Salviati) [reply server/dao/product.js:71]: I've tried but it doesn't work

File: server/dao/product.js (hunk 2 above)
Comment (G-Salviati) [reply server/dao/product.js]: Removed

**Review by MarcoBramini — APPROVED** [review]

**Pull request merged**